### PR TITLE
Fix #459

### LINKED ISSUE #459: Intro slider tutorial for new users
Currently  the app does not contain a intro sliders tutorial at its first run and the features in Zulip may be intimidating for few users. To make sure that every user knows all the features of the Zulip-mobile app There should be a tutorial about all the tools the app provides to increase user’s productivity.

And also having an option to access this tutorials again in future. 

Guidance to new people on react : We can use this [library](https://github.com/FuYaoDe/react-native-app-intro)  for this issue. :)

### PR DESCRIPTION
**Pull request by vihanggodbole**
Added intro slider screen to the app.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@vihanggodbole, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by vihanggodbole** [comment]
Fixing issue #459

**Comment by vihanggodbole** [comment]
@saketkumar95 Thank you! Here - http://imgur.com/a/YgaJi
Btw, could you let me know why are the checks failing?

File: src/ZulipMobile.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import React, { Component } from 'react';
 import { Provider } from 'react-redux';
-
```
Comment (saketkumar95) [inline src/ZulipMobile.js]: Can you remove this change?

File: src/nav/Navigation.js [hunk 2]
```diff
@@ -15,11 +15,11 @@ import SearchMessagesScreen from '../search/SearchMessagesScreen';
 import UsersScreen from '../users/UsersScreen';
 import SubscriptionsScreen from '../subscriptions/SubscriptionsScreen';
 import ChatScreen from '../chat/ChatScreen';
+import TutorialScreen from '../start/TutorialScreen';
 
 const { CardStack: NavigationCardStack } = NavigationExperimental;
 
 export default class Navigation extends React.Component {
-
```
Comment (saketkumar95) [inline src/nav/Navigation.js]: Same as above.

**Comment by saketkumar95** [comment]
I think we need some a option too like 'Take tour' in the app itself to start this tutorial. And one more thing that i think we need to have some more tutorials here related to the hidden features of the app.

**Comment by borisyankov** [comment]
Make sure to run:

```
npm run test
```

before doing the PR. This will uncover issues (mostly formatting in this case).

You can use prettier.js for easier and faster formatting and conforming to ESLint rules.

File: src/nav/routingSelectors.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import { AsyncStorage } from 'react-native';
```
Comment (borisyankov) [inline src/nav/routingSelectors.js]: This is not needed here.

File: src/start/TutorialScreen.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  }
+  nextBtnHandle = (index) => {
+    console.log(index);
+  }
+  onSlideChangeHandle = (index, total) => {
+    console.log(index, total);
+  }
+  
+  render() {
+    const pageArray = [{
```
Comment (borisyankov) [inline src/start/TutorialScreen.js]: Extract this outside the render function.
Even better extract it into a separate module.

**Comment by vihanggodbole** [comment]
@saketkumar95 Yeah that sounds like a good idea! I thought about a simple switch in the settings menu which will show the intro tutorial again to the user. But then there is no settings option so I don't know where to put it. Also, I didn't have any other graphics to use so I couldn't add the small features to the intro.

File: src/nav/Navigation.js (hunk 2 above)
Comment (vihanggodbole) [inline src/nav/Navigation.js]: Will do so.

File: src/nav/routingSelectors.js (hunk 3 above)
Comment (vihanggodbole) [inline src/nav/routingSelectors.js]: I was testing something and I missed it. Sorry!

File: src/start/TutorialScreen.js (hunk 4 above)
Comment (vihanggodbole) [inline src/start/TutorialScreen.js]: I'll try extracting into a separate module. That should be much cleaner.

**Comment by vihanggodbole** [comment]
@borisyankov made all the changes that you requested. I also added the functionality so that the app only shows the tutorial screen once when it is first launch. On subsequent launches it won't show the tutorial screen.

**Comment by saketkumar95** [comment]
@borisyankov @vihanggodbole Can i update this in https://github.com/zulip/zulip-mobile/pull/511 ?

**Comment by vihanggodbole** [comment]
@saketkumar95 sure! I hope this code helps you there.

**Comment by saketkumar95** [comment]
@vihanggodbole Yeah. Thanks. :)

**Comment by kunall17** [comment]
@vihanggodbole Are you working on this? 

Most likely we need screenshots for the Mobile app rather than having screenshots of the web app.

Maybe @saketkumar95 is planning to do this one?

**Comment by borisyankov** [comment]
@vihanggodbole @saketkumar95 this one definitely needs some more discussing before we can move forward. Let's chat on Zulip ;)

**Comment by saketkumar95** [comment]
@kunall17 @borisyankov Yeah , I designed the flow for login , you can check out here https://chat.zulip.org/#narrow/stream/mobile/subject/Login.20flow/near/211739

**Comment by zulipbot-test** [comment]
Hello @vihanggodbole, a Zulip maintainer reviewed this pull request over a week ago, but you haven't updated your pull request since. Please take a look at the requested changes and update your pull request accordingly.

Thank you for your valuable contributions to Zulip!

**Comment by borisyankov** [comment]
Thanks for your efforts. I am closing this for now.
Feel free to reopen, when you have time to work again on it. 👍

**Pull request closed without merging**